src/app/wallet/shared/address-table/address-table.component.ts

```diff
@@ -119,16 +121,29 @@ export class AddressTableComponent implements OnInit {
   /** Delete address */
 
   public deleteAddress(label: string, address: string) {
+    this.address = address;
     if (confirm(`Are you sure you want to delete ${label}: ${address}`)) {
       // this._rpc.call(this, 'manageaddressbook', ['del', address], this.rpc_deleteAddress_success);
-      this._rpc.call('manageaddressbook', ['del', address])
-        .subscribe(response => {
-            this.rpc_deleteAddress_success(response);
-          },
-          error => console.log(`${error.error.message}`));
+      if (this._rpc.state.get('locked')) {
```

Comment:
lets move this manageaddressbook to a local function and get rid of this.address

Reply:
Sorry I forget to remove this.address coz I already send address in the callback :) :) so no need of local function as we can still call it as a callback :)